packages/discovery-provider/src/tasks/index_payment_router.py

```diff
@@ -223,18 +231,34 @@ def get_tx_in_db(session: Session, tx_sig: str) -> bool:
     return exists
 
 
-def parse_route_transaction_memo(
+def parse_route_transaction_memos(
     session: Session, memos: List[str], timestamp: datetime
-) -> RouteTransactionMemo | None:
+) -> Tuple[RouteTransactionMemo | None, GeoMetadataDict | None]:
     """Checks the list of memos for one matching a format of a purchase's content_metadata, and then uses that content_metadata to find the stream_conditions associated with that content to get the price"""
     if len(memos) == 0:
-        return None
+        return None, None
+
+    route_transaction_memo = None
+    geo_memo = None
+
     for memo in memos:
-        if memo == RECOVERY_MEMO_STRING:
-            return RouteTransactionMemo(
-                type=RouteTransactionMemoType.recovery, metadata=None
-            )
         try:
+            if memo == RECOVERY_MEMO_STRING:
+                route_transaction_memo = RouteTransactionMemo(
+                    type=RouteTransactionMemoType.recovery, metadata=None
+                )
+                continue
+            if memo.startswith(GEO_MEMO_STRING):
+                geo_data = json.loads(memo.replace(GEO_MEMO_STRING, ""))
+                geo_memo = GeoMetadataDict(
```

Comment:
Should we do any validation on this? Might be helpful to log errors in cases where the format is invalid or the values are garbage.

Reply:
added a little bit for now, but i think it makes sense for us to accept only geo stuff as a valid ISO format in the future (and update the listen programs too) and do some strict validation. good callout